exploration/04_trigger_exploration.R

```diff
@@ -0,0 +1,851 @@
+#### FIND WORST YEARS ####
+##########################
+
+# worst end of season vegetation health index
+df_vhi_wy <- df_vhi |> 
+  dplyr$filter(
+    Month == "05",
+    Dekad == 3
+  ) |> 
+  dplyr$arrange(
+    Data
+  ) |> 
+  dplyr$slice(
+    1:8
+  ) |> 
+  dplyr$transmute(
+    indicator = "VHI",
+    year = Year
+  )
+
+# look from 2000 onward
+df_vhi_wy_2000 <- df_vhi |> 
+  dplyr$filter(
+    Month == "05",
+    Dekad == 3,
+    Year >= 2000
+  ) |> 
+  dplyr$arrange(
+    Data
+  ) |> 
+  dplyr$slice(
+    1:5
+  ) |> 
+  dplyr$transmute(
+    indicator = "VHI",
+    year = Year
+  )
+
+df_seas5_wy <- df_seas5 |> 
+  dplyr$filter(
+    valid_month %in% 3:5
+  ) |> 
+  dplyr$group_by(
+    date,
+    valid_year
+  ) |> 
+  dplyr$filter(
+    dplyr$n() == 3
+  ) |> 
+  dplyr$summarize(
+    precipitation = sum(precipitation),
+    leadtime = min(leadtime),
+    pub_month = if (leadtime < 3) 3 - leadtime else 15 - leadtime,
+    .groups = "drop"
+  ) |> 
+  dplyr$filter(
+    valid_year >= 1984
+  ) |> 
+  dplyr$group_by(
+    leadtime
+  ) |> 
+  dplyr$arrange(
+    precipitation,
+    .by_group = TRUE
+  ) |>
+  dplyr$slice(
+    1:8
+  ) |> 
+  dplyr$ungroup() |> 
+  dplyr$transmute(
+    indicator = "SEAS5",
+    year = valid_year,
+    leadtime,
+    pub_month
+  )
+
+df_ndvi_wy <- df_ndvi |>
+  dplyr$filter(
+    Month %in% c("03", "04", "05")
+  ) |> 
+  dplyr$group_by(
+    Year
+  ) |> 
+  dplyr$mutate(
+    ndvi_cum = cumsum(Data),
+    ndvi_cum_avg = cumsum(Data_long_term_Average),
+    ndvi_cum_anom = ndvi_cum / ndvi_cum_avg
+  ) |> 
+  dplyr$group_by(
+    Month,
+    Dekad
+  ) |> 
+  dplyr$arrange(
+    ndvi_cum,
+    .by_group = TRUE
+  ) |> 
+  dplyr$slice(
+    1:8
+  ) |> 
+  dplyr$ungroup() |> 
+  dplyr$transmute(
+    indicator = "NDVI",
+    month = as.numeric(Month),
+    dekad = as.numeric(Dekad),
+    year = Year
+  )
+
+# since 2000
+NDVI <- df_ndvi |>
+  dplyr$filter(
+    Month %in% c("03", "04", "05"),
+    Year >= 2000
+  ) |> 
+  dplyr$group_by(
+    Year
+  ) |> 
+  dplyr$mutate(
+    ndvi_cum = cumsum(Data),
+    ndvi_cum_avg = cumsum(Data_long_term_Average),
+    ndvi_cum_anom = ndvi_cum / ndvi_cum_avg
+  ) |> 
+  dplyr$group_by(
+    Month,
+    Dekad
+  ) |> 
+  dplyr$arrange(
+    ndvi_cum,
+    .by_group = TRUE
+  ) |> 
+  dplyr$slice(
+    1:5
+  ) |> 
+  dplyr$ungroup() |> 
+  dplyr$transmute(
+    indicator = "NDVI",
+    month = as.numeric(Month),
+    dekad = as.numeric(Dekad),
+    year = Year
+  ) |> 
+  dplyr$filter(
+    month == 5,
+    dekad == 3
+  )
+
+df_chirps_wy <- df_chirps |> 
+  dplyr$mutate(
+    month = lubridate$month(date),
+    year = lubridate$year(date)
+  ) |> 
+  dplyr$filter(
+    month %in% 1:5,
+    year >= 1984
+  ) |> 
+  dplyr$group_by(
+    year
+  ) |> 
+  dplyr$mutate(
+    precipitation_cum = cumsum(value)
+  ) |>
+  dplyr$group_by(
+    month
+  ) |> 
+  dplyr$arrange(
+    precipitation_cum,
+    .by_group = TRUE
+  ) |> 
+  dplyr$slice(
+    1:8
```

Comment:
comment that this corresponds to ~1 in 5 year RP. It wasn't until I got to the end of the code and came back and really thought about it that i realized this.